### PR TITLE
🐛Prevent autoscaling from creating workers indefinitely

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
- drained node instance resources were computed using docker facilities instead of EC2, thus leading to an infinite creation of EC2 instance.

### Bonus
- fixes time to terminate
- shows logs before terminating in info
- adds a pre-commit hook to check for pytest.mark.testit

## Related issue/s
- fixes https://github.com/ITISFoundation/osparc-simcore/issues/4999

### CODE REVIEW

File: services/autoscaling/tests/unit/test_modules_auto_scaling_computational.py [hunk 1]
```diff
@@ -312,6 +312,7 @@ def _do(
     return _do
 
 
+@pytest.mark.testit
```
Comment (GitHK) [inline services/autoscaling/tests/unit/test_modules_auto_scaling_computational.py:315]: maybe you forgot this one

**Review by GitHK — APPROVED** [review]

File: services/autoscaling/tests/unit/test_modules_auto_scaling_computational.py (hunk 1 above)
Comment (sanderegg) [reply services/autoscaling/tests/unit/test_modules_auto_scaling_computational.py:315]: indeed

**Review by matusdrobuliak66 — APPROVED** [review]

File: scripts/precommit/pytest-testit.bash [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# Check for the presence of @pytest.mark.testit in staged files
+git diff --cached --name-only | while IFS= read -r file; do
+    if grep -n '@pytest\.mark\.testit' "$file"; then
+        echo "Error: Your commit contains '@pytest.mark.testit' in file '$file'. Please remove it before committing."
+        exit 1
+    fi
+done
```
Comment (bisgaard-itis) [inline scripts/precommit/pytest-testit.bash:9]: Nice!

**Review by bisgaard-itis — APPROVED** [review]
So, for my understanding: Are these changes related to the issues you mentioned to me were detected by the e2e tests I set up?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -45,3 +45,10 @@ repos:
     hooks:
       - id: black
         name: black format code
+  - repo: local
+    hooks:
+    - id: pytest-testit
+      name: pytest-testit
+      language: script
+      types: [file, python]
+      entry: scripts/precommit/pytest-testit.bash
```
Comment (pcrespov) [inline .pre-commit-config.yaml:54]: hehe I feel a bit guilty for having added thisone

AFAIK these scripts act only on changed files, right? If that is the case, that approach can be useful to do some checks on requirements files that I cannot always enforce by code (e.g. do not install `respx` in base requreiments). I have added a standard test for it but I think it makes more sense to have it here (or perahps not?)

File: scripts/precommit/pytest-testit.bash [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# Check for the presence of @pytest.mark.testit in staged files
+git diff --cached --name-only | while IFS= read -r file; do
+    if grep -n '@pytest\.mark\.testit' "$file"; then
+        echo "Error: Your commit contains '@pytest.mark.testit' in file '$file'. Please remove it before committing."
```
Comment (pcrespov) [inline scripts/precommit/pytest-testit.bash:6]: Error message should be "Error; Pedro, again? Your commit contains ... "

**Comment by sanderegg** [comment]
> So, for my understanding: Are these changes related to the issues you mentioned to me were detected by the e2e tests I set up?

@bisgaard-itis : no this is another issue. the one of the e2e I am currently working on it

File: services/autoscaling/src/simcore_service_autoscaling/models.py [hunk 5]
```diff
@@ -44,7 +44,7 @@ def __sub__(self, other: "Resources") -> "Resources":
 
 @dataclass(frozen=True)
 class EC2InstanceType:
-    name: str
+    name: InstanceTypeType
```
Comment (pcrespov) [inline services/autoscaling/src/simcore_service_autoscaling/models.py:47]: OK this is the second time this morning. `RequestRequest` and not `TypeType`? 
Please a bit of imagination with names ! :-)
Aha this comes from `types_aiobotocore_ec2.literals`!! OMG! :-D

**Review by pcrespov — APPROVED** [review]
thxthx! ;-)

File: services/autoscaling/src/simcore_service_autoscaling/models.py (hunk 5 above)
Comment (sanderegg) [reply services/autoscaling/src/simcore_service_autoscaling/models.py:47]: yes. complain to Amazon...

File: .pre-commit-config.yaml (hunk 3 above)
Comment (sanderegg) [reply .pre-commit-config.yaml:54]: might be a thing

**Pull request merged**